File: src/Java/adventofcode2018/Day11.java
package adventofcode2018;

/**
 * https://adventofcode.com/2018/day/11
 */
public class Day11 {

    static void findMax(int input, int s1, int s2) {
        int max = -1000000;
        int maxx = 0, maxy = 0, maxs = 0;
        for (int size = s1; size<=s2; size++) {
            for (int i=1; i<=298; i++) {
                for (int j=1; j<=298; j++) {
                    int s = Math.min(size, Math.min(301-i, 301-j));
                    int power = 0;
                    for (int x = 0; x<s; ++x) {
                        for (int y = 0; y<s; ++y) {
                            int rackID = (i+x) + 10;
                            int p = (rackID * (y+j) + input) * rackID;
                            power += (p/100) % 10 - 5;
                        }
                    }
                    if (max < power) {
                        max = power;
                        maxx = i;
                        maxy = j;
                        maxs = s;
                    }
                }
            }
        }
        if (s1 == s2) {
            System.err.printf("%d,%d %d\n", maxx, maxy, max);
        } else {
            System.err.printf("%d,%d,%d %d\n", maxx, maxy, maxs, max);
        }
    }

    public static void main(String[] args) {
        int input = 3613;
        findMax(input, 3, 3);
        findMax(input, 1, 30);
    }

}

File: src/Python/2018/Day11.py

from utils import Input, printResult

# https://adventofcode.com/2018/day/11

def getPower(x, y, serial):
    id = x + 10
    return (((id * y + serial) * id) // 100) % 10 - 5

serial = 3613
stripes = []
for i in range(300):
    stripes.append([[] for _ in range(300)])
    for j in range(300):
        power = 0
        for s in range(min(30, 300-j)):
            power += getPower(i+1, j+s+1, serial)
            stripes[i][j].append(power)

def findMax(s1, s2):
    maxPower = 0
    maxx, maxy, maxs = 0, 0, 0
    for size in range(s1, s2+1):
        for i in range(301-size):
            for j in range(301-size):
                power = sum(stripes[i+s][j][size-1] for s in range(size))
                if maxPower < power:
                    maxPower, maxx, maxy, maxs = power, i, j, size
    return (maxx+1, maxy+1, maxs, maxPower)

part1 = findMax(3, 3)
printResult(1, "(%d,%d) %d" % (part1[0], part1[1], part1[2]))
part2 = findMax(1, 30)
printResult(2, "(%d,%d,%d) %d" % part2)
